### PR TITLE
CLN/TST: address TODOs

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Review by jreback — APPROVED** [review]

File: pandas/core/series.py [hunk 1]
```diff
@@ -4528,6 +4528,9 @@ def rename(
         5    3
         dtype: int64
         """
+        if axis is not None:
+            axis = self._get_axis_number(axis)
```
Comment (MarcoGorelli) [inline pandas/core/series.py:4532]: @jbrockmendel it doesn't seem like `axis` is used in this function - why is this necessary?

Comment (jbrockmendel) [reply pandas/core/series.py:4532]: so that we raise if the user passes something wacky https://github.com/pandas-dev/pandas/pull/44211/files#diff-305bb1e81d72a15c06b848966eb339db5a0546da4e7a17e1afab7cc9d545510eL71

Comment (MarcoGorelli) [reply pandas/core/series.py:4532]: ah, makes sense - thanks for explaining!

Comment (jbrockmendel) [reply pandas/core/series.py:4532]: sure thing.  will also add a comment to this effect in an upcoming CLN branch
